packages/cli/scripts/portalClient.ts

```diff
@@ -0,0 +1,171 @@
+  const privateKey = await keys.generateKeyPair('secp256k1')
+  const enr = SignableENR.createFromPrivateKey(privateKey)
+  const nodeAddr = multiaddr(`/ip4/127.0.0.1/udp/${port}`)
+  enr.setLocationMultiaddr(nodeAddr)
+
+  const node = await UltralightProvider.create({
+    transport: TransportLayer.NODE,
+    supportedNetworks: [
+      { networkId: NetworkId.HistoryNetwork },
+      { networkId: NetworkId.StateNetwork },
+    ],
+    config: {
+      enr,
+      bindAddrs: { ip4: nodeAddr },
+      privateKey,
+    },
+
+  })
+
+  return node
+}
+
+async function sendNetworkMessage(node: UltralightProvider, networkId: NetworkId, messageType: number, payload: any = {}): Promise<any> {
+  console.log(`Sending message type ${messageType} to network ${networkId}:`, payload) 
+
+  await new Promise(resolve => setTimeout(resolve, 5000))
+
+  const serializedPayload = new TextEncoder().encode(JSON.stringify({
+    type: messageType,
+    ...payload
+  }))
+
+  try {
+    const response = await node.portal.sendPortalNetworkMessage(
+      node.portal.discv5.enr.toENR(),
+      serializedPayload,
+      networkId
+    ) 
+    return response
+  } catch (error) {
+    console.error('Failed to send network message:', error)
+    throw error
+  }
+}
+
+async function executeMethod(node: UltralightProvider, method: string, params: any[]) {
+  try {
+    const [prefix, methodName] = method.split('_')
+    
+    if (prefix === 'portal') {
+
+      if (methodName === 'statePing') {
+        return await sendNetworkMessage(node, NETWORK_IDS.STATE as NetworkId, MESSAGE_TYPES.PING)
+      } else if (methodName === 'historyPing') {
+        return await sendNetworkMessage(node, NETWORK_IDS.HISTORY as NetworkId, MESSAGE_TYPES.PING)
+      }
+
+      const historyNetwork = node.portal.network()[NetworkId.HistoryNetwork]
+      const stateNetwork = node.portal.network()[NetworkId.StateNetwork]
+      
+      if (historyNetwork && methodName.startsWith('history')) {
+        const networkMethod = methodName.replace('history', '').toLowerCase()
+        if (typeof historyNetwork[networkMethod] === 'function') {
+          return await historyNetwork[networkMethod](...params)
+        }
+      }
+
+      if (stateNetwork && methodName.startsWith('state')) {
+        const networkMethod = methodName.replace('state', '').toLowerCase()
+        if (typeof stateNetwork[networkMethod] === 'function') {
+          return await stateNetwork[networkMethod](...params)
+        }
+      }
+
+      if (typeof node.portal[methodName] === 'function') {
+        return await node.portal[methodName](...params)
+      }
+      
+      throw new Error(`Unknown method: ${methodName}`)
+    }
+
+    throw new Error(`Invalid method prefix: ${prefix}. Must be 'portal'`)
+  } catch (error) {
+    console.error('Error executing method:', error)
+    throw error
+  }
+}
+
+async function main() {
+  let node: UltralightProvider | undefined
```

Comment:
Let's replace this with a pure `PortalNetwork` client instead of the `UltralightProvider` since you're not accessing the provider methods.